test/unit/controllers/actionbuilders/AuthActionSpec.scala

```diff
@@ -71,17 +71,17 @@ class AuthActionSpec extends UnitSpec with MockitoSugar {
     "authorise as CSP when authorised by auth API and both badge identifier and submitter headers exist" in new SetUp {
       authoriseCsp()
 
-      private val actual: AuthorisedRequest[AnyContentAsXml] = await(authAction.refine(request(testFakeRequestWithMaybeBadgeIdAndMaybeSubmitterId())).right.get)
+      private val actual: AuthorisedRequest[AnyContentAsXml] = await(authAction.refine(request(testFakeRequestWithMaybeBadgeIdAndMaybeSubmitterId())).getOrElse(throw new RuntimeException("failed to refine test authAction")))
```

Comment:
I'd prefer to use `inside` (https://www.scalatest.org/user_guide/using_inside) to pattern match on a `Right` to separate the assertion logic from the logic of the system under test.

Reply:
Fair enough - different strokes.